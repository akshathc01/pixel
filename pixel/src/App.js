import './App.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import React, { createContext, useState } from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from "react-router-dom";
import Home from './components/Home/Home/Home';
import AddImage from './components/AddImage/AddImage';
import Login from './components/Login/Login';
import Dashboard from './components/Dashboard/Dashboard';
import PrivateRoute from './components/PrivateRoute/PrivateRoute';

export const UserContext = createContext();
function App() {
  const [loggedInUser, setLoggedInUser] = useState({});
  return (
    <div data-testid="test-App">
      <UserContext.Provider value={[loggedInUser, setLoggedInUser]} >
        <Router>
          <Switch>
            <Route exact path="/">
              <Home />
            </Route>
            <PrivateRoute path="/addImage">
              <AddImage />
            </PrivateRoute>
            <PrivateRoute path="/dashBoard">
              <Dashboard />
            </PrivateRoute>
            <Route path="/signedIn">
              <Login />
            </Route>
          </Switch>
        </Router>
      </UserContext.Provider>
    </div>

  );
}

export default App;
