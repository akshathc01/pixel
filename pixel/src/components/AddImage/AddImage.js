import React, { useContext, useState } from 'react';
import './AddImage.css';
import Navbar from '../Home/Navbar/Navbar';
import { storage } from '../../firebase';
import { UserContext } from '../../App';
import Swal from 'sweetalert2';


const AddImage = () => {
  const [loggedInUser, setLoggedInUser] = useContext(UserContext);
  console.log("Beacuse Test", loggedInUser);

  const [title, setTitle] = useState('')
  const [image, setImage] = useState(null);
  const [url, setUrl] = useState("");
  const [progress, setProgress] = useState(0);
  var storageRef = storage.ref();

  const handleChange = e => {
    if (e.target.files[0]) {
      setImage(e.target.files[0]);
    }
  };

  const handleUpload = async () => {
    const uploadTask = storage.ref(`images/${image.name}`).put(image);
    uploadTask.on(
      "state_changed",
      snapshot => {
        const progress = Math.ceil(
          (snapshot.bytesTransferred / snapshot.totalBytes) * 100
        );
        setProgress(progress);
      },
      error => {
        // console.log(error);
      },
      () => {
        storage
          .ref("images")
          .child(image.name)
          .getDownloadURL()
          .then((url) => {
            setUrl(url);
          });
      }
    );


    const gettingFormPart = document.getElementById('uploadImage');
    const hidingImagePart = document.getElementById('imagePart');
    gettingFormPart.style.display = 'none';
    hidingImagePart.style.display = 'block';
  };

  const handleDelete = () => {
    var desertRef = storageRef.child(`images/${image.name}`);
    desertRef.delete().then(() => {
      Swal.fire({
        position: 'center',
        icon: 'success',
        title: 'Your image has been deleted',
        showConfirmButton: false,
        timer: 1500
      })
    }).catch((error) => {
      // console.log(error);
    });
  }

  const handlePrivate = () => {
    fetch('http://localhost:3002/addPrivate', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        imageName: image.name,
        url,
        email: loggedInUser.email,
        title
      })
    })
    Swal.fire({
      position: 'center',
      icon: 'success',
      title: 'Your image has been uploaded',
      showConfirmButton: false,
      timer: 1500
    })
  }

  const handleCommunity = () => {
    fetch('http://localhost:3002/addCommunity', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify({
        imageName: image.name,
        url,
        email: loggedInUser.email,
        title
      })

    })
    Swal.fire({
      position: 'center',
      icon: 'success',
      title: 'Your image has been uploaded',
      showConfirmButton: false,
      timer: 1500
    })
  }

  return (
    <section data-testid="test-addImage">
      <Navbar />
      <div style={{ position: "relative", paddingTop: '50px', background: "#12161f", height: "910px" }} className="addImage">
        <div
          style={{
            position: "absolute",
            top: "50%",
            left: "50%",
            transform: "translate(-50%, -50%)"
          }}
          className="container text-center">


        </div>

        <form id="uploadImage" className="container">
          <h1 className="text-center text-light pb-5">Please Click On The Button To Upload Your Image</h1>
          <div className="form-row d-flex justify-content-center">
            <div className="input-group input-group-lg col-6">
              <input onChange={handleChange} type="file" className="form-control pt-1" aria-label="Sizing example input" aria-describedby="inputGroup-sizing-lg" />
            </div>

            <div className="col-1 ">
              <button style={{ padding: '11px' }} onClick={handleUpload} type="button" className="btn btn-danger">Upload</button>
            </div>
          </div>
        </form>

        <div style={{ display: 'none' }} id="imagePart" className=" my-3 container text-center">
          <h6 className="text-light py-3 pb-5"> Image URL: {url} </h6>
          <img src={url} className="card-img-top text-center pb-5" alt={title} style={{ width: '30%', height: 'auto' }} />
          <div className="row pb-5">
            <div className="input-group my-3 w-50 text-center col-10 ">
              <input onChange={(e) => { setTitle(e.target.value) }} type="text" className="form-control border border-danger" placeholder="Image Title" aria-label="Image Title" aria-describedby="button-addon2" />

              <button onClick={handlePrivate} className="btn btn-outline-danger text-light ml-3" type="button" id="button-addon2">Add Private</button>
              <button onClick={handleCommunity} className="btn btn-outline-danger ml-3 col-2 text-light" type="button"> To Community </button>
              <button onClick={handleDelete} className="btn btn-outline-danger ml-3 col-2 text-light" type="button"> Delete </button>
            </div>
          </div>
        </div>
      </div>
    </section>
  );
};

export default AddImage;