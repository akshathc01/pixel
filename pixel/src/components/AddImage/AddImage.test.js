import { render, screen, cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { useContext } from 'react'
import { UserContext } from '../../App'
import AddImage from './AddImage'
import { BrowserRouter as Router } from 'react-router-dom';

test('test', () => {
    expect(true).toBe(true)
})

test('it will render AddImage component', () => {
    const loggedInUser = {
        email: "eusha37@gmail.com",
        name: "Mohammad Eusha",
        isSignedIn: true,
        photo: "https://lh3.googleusercontent.com/a-/AOh14GhyyoLqA1sJPhj7Xib2QoaPG1-KARKEMhcQSoYpDA=s96-c"
    }

    render(
        <UserContext.Provider value={[loggedInUser]}>
            <Router> <AddImage /> </Router>
        </UserContext.Provider>

    )
    const addImageElement = screen.getByTestId('test-addImage')
    expect(addImageElement).toBeInTheDocument()
})

test('matches snapshot for AddImage', () => {
    const loggedInUser = {
        email: "eusha37@gmail.com",
        name: "Mohammad Eusha",
        isSignedIn: true,
        photo: "https://lh3.googleusercontent.com/a-/AOh14GhyyoLqA1sJPhj7Xib2QoaPG1-KARKEMhcQSoYpDA=s96-c"
    }
    const tree = renderer.create(
        <UserContext.Provider value={[loggedInUser]}>
            <Router> <AddImage /> </Router>
        </UserContext.Provider>
    ).toJSON()
    expect(tree).toMatchSnapshot()
})