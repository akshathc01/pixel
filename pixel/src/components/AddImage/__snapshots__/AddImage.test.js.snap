// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for AddImage 1`] = `
Array [
  " ",
  <section
    data-testid="test-addImage"
  >
    <div
      className="full-container"
      data-testid="test-Navbar"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light p-0 bg-dark "
      >
        <div
          className="container-fluid text-light"
        >
          <a
            className="navbar-brand d-flex align-items-center"
            href="/"
            onClick={[Function]}
          >
            <img
              alt="Icon"
              className="logo"
              src="icon.png"
              style={
                Object {
                  "width": "50px",
                }
              }
            />
            <h1
              className="pl-3 text-light"
            >
              Pixel
            </h1>
          </a>
          <button
            className="btn btn-danger ml-3"
          >
            <a
              className="text-light"
              href="/addImage"
              onClick={[Function]}
            >
               Upload Image 
            </a>
          </button>
          <button
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
            className="navbar-toggler"
            data-target="#navbarNavAltMarkup"
            data-toggle="collapse"
            type="button"
          >
            <span
              className="navbar-toggler-icon"
            />
          </button>
          <div
            className="collapse navbar-collapse justify-content-end"
            id="navbarNavAltMarkup"
          >
            <div
              className="navbar-nav"
            >
              <a
                className="nav-link active px-3 text-light h5  my-2"
                href="/"
                onClick={[Function]}
              >
                 Home 
                <span
                  className="sr-only"
                >
                  (current)
                </span>
                 
              </a>
              <a
                className="nav-link px-3 my-2"
              >
                <a
                  className="text-light h5 "
                  href="/dashboard"
                  onClick={[Function]}
                >
                  Images
                </a>
              </a>
              <button
                className="btn btn-outline-danger mx-3 my-2"
                type="button"
              >
                 
                <a
                  className="text-light"
                  href="/signedIn"
                  onClick={[Function]}
                >
                  Sign In
                </a>
                  
              </button>
              <button
                className="btn btn-danger mx-3 my-2"
              >
                <a
                  className="text-light"
                  href="/signedIn"
                  onClick={[Function]}
                >
                  Sign Up
                </a>
              </button>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div
      className="addImage"
      style={
        Object {
          "background": "#12161f",
          "height": "910px",
          "paddingTop": "50px",
          "position": "relative",
        }
      }
    >
      <div
        className="container text-center"
        style={
          Object {
            "left": "50%",
            "position": "absolute",
            "top": "50%",
            "transform": "translate(-50%, -50%)",
          }
        }
      />
      <form
        className="container"
        id="uploadImage"
      >
        <h1
          className="text-center text-light pb-5"
        >
          Please Click On The Button To Upload Your Image
        </h1>
        <div
          className="form-row d-flex justify-content-center"
        >
          <div
            className="input-group input-group-lg col-6"
          >
            <input
              aria-describedby="inputGroup-sizing-lg"
              aria-label="Sizing example input"
              className="form-control pt-1"
              onChange={[Function]}
              type="file"
            />
          </div>
          <div
            className="col-1 "
          >
            <button
              className="btn btn-danger"
              onClick={[Function]}
              style={
                Object {
                  "padding": "11px",
                }
              }
              type="button"
            >
              Upload
            </button>
          </div>
        </div>
      </form>
      <div
        className=" my-3 container text-center"
        id="imagePart"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <h6
          className="text-light py-3 pb-5"
        >
           Image URL: 
          
           
        </h6>
        <img
          alt=""
          className="card-img-top text-center pb-5"
          src=""
          style={
            Object {
              "height": "auto",
              "width": "30%",
            }
          }
        />
        <div
          className="row pb-5"
        >
          <div
            className="input-group my-3 w-50 text-center col-10 "
          >
            <input
              aria-describedby="button-addon2"
              aria-label="Image Title"
              className="form-control border border-danger"
              onChange={[Function]}
              placeholder="Image Title"
              type="text"
            />
            <button
              className="btn btn-outline-danger text-light ml-3"
              id="button-addon2"
              onClick={[Function]}
              type="button"
            >
              Add Private
            </button>
            <button
              className="btn btn-outline-danger ml-3 col-2 text-light"
              onClick={[Function]}
              type="button"
            >
               To Community 
            </button>
            <button
              className="btn btn-outline-danger ml-3 col-2 text-light"
              onClick={[Function]}
              type="button"
            >
               Delete 
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>,
  " ",
]
`;
